app/percentage_controller.py

```diff
@@ -2,22 +2,31 @@
 import maya
 import cgi
 import qq
+from .models import User, Update, DBSession
 
 percentage = Blueprint('percentage', __name__)
 
 @percentage.route("/percentage", methods=['POST'])
 def percentage_update():
   data = request.get_json()
-  username = cgi.escape(data["username"])
+  zap_token = data["zap_token"]
+
   try:
     percentage = int(data["percentage"])
-    if percentage < 0 or percentage > 100 or len(username) > 16:
+    if percentage < 0 or percentage > 100 or len(zap_token) > 64:
       return "Reported percentage out of bounds", 400
 
     charging = bool(data["charging"])
     updated_at = maya.now().datetime(to_timezone='US/Eastern')
 
-    qq.insert_update(username, percentage, charging, updated_at)
+    with DBSession() as session:
```

Comment:
There are so many session.query(User)...first() we could abstract this logic maybe

Reply:
Maybe later